src/screens.web.js

```diff
@@ -1,19 +1,110 @@
-import { Animated, View } from 'react-native';
+import debounce from 'debounce';
+import React from 'react';
+import { Animated, StyleSheet, View } from 'react-native';
+
+let _shouldUseScreens = true;
 
 export function useScreens(shouldUseScreens = true) {
   if (shouldUseScreens) {
-    console.warn('react-native-screens is not support on this platform.');
+    console.warn(
+      'react-native-screens is not fully support on this platform yet.'
+    );
   }
+  _shouldUseScreens = shouldUseScreens;
 }
 
 export function screensEnabled() {
-  return false;
+  return _shouldUseScreens;
+}
+
+function isAnimatedValue(value) {
+  return value && value.__getValue && value.addListener;
+}
+
+function isPropTruthy(prop) {
+  let activeValue = prop;
+  if (isAnimatedValue(prop)) {
+    activeValue = !!prop.__getValue();
+  }
+
+  return activeValue !== undefined && activeValue;
 }
 
-export const Screen = Animated.View;
+export class Screen extends React.Component {
+  listenerId = null;
+
+  state = { isActive: null };
+
+  constructor(props) {
+    super(props);
+
+    this._onAnimatedValueUpdated = debounce(this._onAnimatedValueUpdated, 10);
+
+    this._addListener(props.active);
+
+    this.state = {
+      isActive: isPropTruthy(props.active),
+    };
+  }
+
+  componentWillUnmount() {
+    this._removeListener(this.props.active);
+  }
+
+  _addListener = possibleListener => {
+    if (isAnimatedValue(possibleListener)) {
+      this.listenerId = possibleListener.addListener(
+        this._onAnimatedValueUpdated
+      );
+    }
+  };
+
+  _removeListener = possibleListener => {
+    if (isAnimatedValue(possibleListener)) {
+      possibleListener.removeListener(this.listenerId);
+      this.listenerId = null;
+    }
+  };
+
+  componentWillReceiveProps({ active: nextActive }) {
+    const { active } = this.props;
+    if (nextActive !== active) {
+      this._removeListener(active);
+      this._addListener(nextActive);
+
+      this.setState({
+        isActive: isPropTruthy(nextActive),
+      });
+    }
+  }
+
+  _onAnimatedValueUpdated = ({ value }) => {
+    if (this.state.isActive !== !!value) {
+      this.setState({ isActive: !!value });
+    }
+  };
+
+  render() {
+    const { style, ...rest } = this.props;
+    const { isActive } = this.state;
+    let viewStyle = style;
+
+    if (!isActive) {
+      viewStyle = [style, styles.none];
```

Comment:
is there no way we can control `display` prop other than re-rendering? Would it be possible to have `display = new Animated.Value('block')` and then set it with `display.setValue('none')`?

I'm pretty certain that with native driver only numeric values are supported with animated, but on web we don't use native driver so maybe we could do this? It would simplify this component a bit as we wouldn't need to use state at all.

What do you think?

Reply:
So I would hijack incoming animation values and update the new display animation value with their value? Wouldn't it also need to observe if the incoming value was just truthy?

Reply:
> So I would hijack incoming animation values and update the new display animation value with their value?

Sort of, but I don't think hijack is the right word since the parent of this component expects it to do whatever it wants with the `isActive`prop. This component's `isActive` prop can either be a regular value or an AnimatedValue, and in the latter case, this component subscribes to changes in the AnimatedValue. In either case, this component would trigger a separate animation that "animates" `block` to `none` immediately.

I put "animates" in quotes because this second, internal AnimatedValue is not used for animation. It is used because AnimatedValues bypass React's rendering and reconciliation. TBH something like `setNativeProps` would also work except for the fact that I'm not sure it handles StyleSheet values that could be in `this.props.style`.

Reply:
do you want to try this other approach @EvanBacon ? otherwise this looks fine, just wanted to see if this implementation could be less complicated

Reply:
AFAICT the `Animated.Value` in react-native-web only animates numeric values.